src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+const getUserById = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const listTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getNewPostId = (todos: Todo[]) => {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(listTodos);
+  const [title, setTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const handleChangeUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setIsUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const onAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newUser = getUserById(userId) || null;
+
+    let hasError = false;
+
+    if (!title) {
+      setIsTitleError(true);
+      hasError = true;
+    }
+
+    if (!userId) {
+      setIsUserError(true);
+      hasError = true;
+    }
+
+    if (hasError) {
+      return;
+    }
+
+    const newTodo = {
+      id: getNewPostId(todos),
+      title,
+      completed: false,
+      userId,
+      user: newUser,
+    };
+
+    setTodos(prevTodos => [...prevTodos, newTodo]);
```

Comment:
There is a syntax error in the `setTodos` function. The spread operator is incorrectly used. Ensure that you are correctly spreading the previous todos array.